0.13.0 release candidate 1 (#2107)
open-quantum-safe/liboqs: RELEASE.md

Release notes
=============

This is version 0.13.0-rc1 of liboqs. It was released on March 19, 2025.

> This release improves support for NIST Additional Signatures Round 2 candidates: CROSS and MAYO implementations are updated and support is added for UOV. This release also adds a new KEM API for deterministic key generation and encapsulation (only supported by ML-KEM at the moment).

Reviewer: ```suggestion
This release improves support for NIST Additional Signatures Round 2 candidates: CROSS and MAYO implementations are updated and support is added for UOV. This release also adds a new KEM API for deterministic key generation (only supported by ML-KEM at the moment).
```
We actually didn't ship deterministic encapsulation, just deterministic key generation.

What's New
----------

This release continues from the 0.12.0 release of liboqs.

### Key encapsulation mechanisms

> - New API: Added a deterministic key generation and encapsulation API for KEMs (only ML-KEM supported at the moment).

Reviewer: ```suggestion
- New API: Added a deterministic key generation API for KEMs (only ML-KEM supported at the moment).
```

- ML-KEM: Added support for the ML-KEM implementation from [Nvidia cuPQC](https://developer.nvidia.com/cupqc), a GPU accelerated cryptography library.

> - ML-KEM: Implementation from mlkem-native upstream updated to add Pair-wise Consistency Test (PCT) and Intel CET support.

Reviewer 2: I believe the mlkem-native implementation is new for this release. May be worth an extra bullet point.

* Add checks for ML-KEM keys by @abhinav-thales in https://github.com/open-quantum-safe/liboqs/pull/2009
* Update actions/cache to v4.2.2 by @mkannwischer in https://github.com/open-quantum-safe/liboqs/pull/2093
* Add Nix flake by @aidenfoxivey in https://github.com/open-quantum-safe/liboqs/pull/1970
* Update MAYO to NIST round 2 by @bhess in https://github.com/open-quantum-safe/liboqs/pull/2095
* Update mlkem-native to v1.0.0-beta by @mkannwischer in https://github.com/open-quantum-safe/liboqs/pull/2092
* Add references to security response process by @SWilson4 in https://github.com/open-quantum-safe/liboqs/pull/2077
* Bump version to 0.13.0-dev [skip ci] by @SWilson4 in https://github.com/open-quantum-safe/liboqs/pull/2099
* Add UOV  by @mkannwischer in https://github.com/open-quantum-safe/liboqs/pull/2094
* Add bitflip test for trivial SUF-CMA forgeries by @rtjk in https://github.com/open-quantum-safe/liboqs/pull/2090
* Update MAYO version in algorithm datasheet by @bhess in https://github.com/open-quantum-safe/liboqs/pull/2103
* Add DeriveKeyPair API by @SWilson4 in https://github.com/open-quantum-safe/liboqs/pull/2070

> * Update nist-round in UOV and MAYO data sheet by @bhess in https://github.com/open-quantum-safe/liboqs/pull/2105

Reviewer: It looks like the changelog line for #2104 is missing.

## New Contributors
* @zhaixiaojuan made their first contribution in https://github.com/open-quantum-safe/liboqs/pull/2010
* @stevenireeves made their first contribution in https://github.com/open-quantum-safe/liboqs/pull/2044
* @pablo-gf made their first contribution in https://github.com/open-quantum-safe/liboqs/pull/2059
* @levitte made their first contribution in https://github.com/open-quantum-safe/liboqs/pull/2086

> * @mkannwischer made their first contribution in https://github.com/open-quantum-safe/liboqs/pull/2093

Reviewer: Adding a line for #2104 will also add a first-time contributor here.

Release notes
=============

This is version 0.13.0-rc1 of liboqs. It was released on March 19, 2025.

This release improves support for NIST Additional Signatures Round 2 candidates: CROSS and MAYO implementations are updated and support is added for UOV. This release also adds a new KEM API for deterministic key generation (only supported by ML-KEM at the moment). Finally, this release adds support for ML-KEM implementations from 2 new sources: formally verified portable C, AVX2, and AArch64 implementations from [PQCP's mlkem-native](https://github.com/pq-code-package/mlkem-native) and a GPU accelerated CUDA implementation from [Nvidia cuPQC](https://developer.nvidia.com/cupqc). 

What's New
----------

This release continues from the 0.12.0 release of liboqs.

### Key encapsulation mechanisms

- New API: Added a deterministic key generation and API for KEMs (only ML-KEM supported at the moment).

> - ML-KEM: Added support the portable C, AVX2, and AArch64 ML-KEM implementations from [PQCP's mlkem-native](https://github.com/pq-code-package/mlkem-native). Large parts of these implementations are formally verified: all of the C code is verified using [CBMC](https://github.com/diffblue/cbmc) and the functional correctness of the core AArch64 assembly routines is verified using [HOL-Light](https://github.com/jrh13/hol-light). 

Reviewer: Nit:
```suggestion
- ML-KEM: Added support for the portable C, AVX2, and AArch64 ML-KEM implementations from [PQCP's mlkem-native](https://github.com/pq-code-package/mlkem-native). Large parts of these implementations are formally verified: all of the C code is verified using [CBMC](https://github.com/diffblue/cbmc) and the functional correctness of the core AArch64 assembly routines is verified using [HOL-Light](https://github.com/jrh13/hol-light). 
```

> - ML-KEM: Added support the portable C, AVX2, and AArch64 ML-KEM implementations from [PQCP's mlkem-native](https://github.com/pq-code-package/mlkem-native). Large parts of these implementations are formally verified: all of the C code is verified using [CBMC](https://github.com/diffblue/cbmc) and the functional correctness of the core AArch64 assembly routines is verified using [HOL-Light](https://github.com/jrh13/hol-light). 

Reviewer 3: It may be worth mentioning that this is a change in the default.
```suggestion
- ML-KEM: Changed the default ML-KEM implementation to [PQCP's mlkem-native](https://github.com/pq-code-package/mlkem-native). There are three variants: Portable C, AVX2, and AArch64. Large parts of these implementations are formally verified: all of the C code is verified for memory and type safety using [CBMC](https://github.com/diffblue/cbmc) and the functional correctness of the core AArch64 assembly routines is verified using [HOL-Light](https://github.com/jrh13/hol-light). 
```

Author: Thank you for the suggestion @hanno-becker, I've added it.
